src/main/kotlin/com/vk/admstorm/ssh/SshHandler.kt

```diff
@@ -0,0 +1,26 @@
+package com.vk.admstorm.ssh
+
+import com.intellij.ssh.SshException
+import java.util.concurrent.TimeoutException
+
+object SshHandler {
+    inline fun <reified T> handle(call: () -> T): T {
+        return try {
+            call()
+        } catch (e: SshException) {
+            throw unwrap(e)
+        }
+    }
+
+    fun unwrap(ex: Exception): Throwable {
+        var cause = ex.cause
+        while (cause?.cause != null) {
+            cause = cause.cause
+        }
+
+        if (cause is TimeoutException) {
```

Comment:
Let's not do it this way. It is better to move the check to `catch` where you call `SshHandler.handle`